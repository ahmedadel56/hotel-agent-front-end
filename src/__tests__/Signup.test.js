import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import Signup from '../components/SignUp';

it('Check if Sign up page renders correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Signup />
      </BrowserRouter>
    </Provider>,
  );
  expect(tree).toMatchSnapshot();
});

afterEach(() => {
  jest.clearAllMocks();
});
