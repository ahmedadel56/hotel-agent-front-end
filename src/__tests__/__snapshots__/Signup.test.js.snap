// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Sign up page renders correctly 1`] = `
<div
  className="reserve-container"
  style={
    Object {
      "backgroundImage": "url(authentication.jpg)",
    }
  }
>
  <div
    className="reserve-heading"
  >
    <h1>
      CREATE AN ACCOUNT
    </h1>
    <hr />
  </div>
  <form
    onSubmit={[Function]}
  >
    <div
      className="login-form"
    >
      <input
        maxLength={20}
        minLength={3}
        name="name"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="👤  name"
        required={true}
      />
    </div>
    <div
      className="login-form"
    >
      <input
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="✉️  email"
      />
    </div>
    <div
      className="login-form"
    >
      <input
        minLength={6}
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        placeholder="🔒  password"
        required={true}
        type="password"
      />
    </div>
    <div
      className="login-btn"
    >
      <input
        type="submit"
        value="SIGN UP"
      />
    </div>
  </form>
</div>
`;
